account_payment_batch_process/wizard/account_payment_register.py

```diff
@@ -0,0 +1,563 @@
+            # Update with payment diff data
+            inv_val = self.get_payment_invoice_value(name, line)
+            group_data[partner_id]["inv_val"].update({line.invoice_id.id: inv_val})
+        else:
+            # calculate amount in words
+            check_amount_in_words = self.total_amount_in_words(line, 0)
+            # prepare name
+            self.update_group_pay_data(
+                partner_id, group_data, line, check_amount_in_words
+            )
+
+    def _reconcile_open_invoices(
+        self,
+        line,
+        inv,
+        amount_residual,
+        amount_residual_currency,
+        reconciled,
+        amount,
+        debit_amount_currency,
+        credit_amount_currency,
+    ):
+        acc_part_recnc_obj = self.env["account.partial.reconcile"]
+        line.update(
+            {
+                "amount_residual": amount_residual,
+                "amount_residual_currency": amount_residual_currency,
+                "reconciled": reconciled,
+            }
+        )
+        if inv.move_type == "out_invoice":
+            part_rec_domain = [("debit_move_id", "=", line.id)]
+        elif inv.move_type == "in_invoice":
+            part_rec_domain = [("credit_move_id", "=", line.id)]
+        partial = acc_part_recnc_obj.search(part_rec_domain, limit=1)
+        if partial:
+            partial.update(
+                {
+                    "amount": amount,
+                    "debit_amount_currency": debit_amount_currency,
+                    "credit_amount_currency": credit_amount_currency,
+                }
+            )
+
+    def make_payments(self):
+        # Make group data either for Customers or Vendors
+        context = dict(self._context or {})
+        group_data = {}
+        memo = self.communication or " "
+        context.update({"is_customer": self.is_customer})
+        self._check_amounts()
+        for invoice_payment_line in self.invoice_payments:
+            if invoice_payment_line.amount > 0:
+                self.get_amount(memo, group_data, invoice_payment_line)
+        # update context
+        context.update({"group_data": group_data})
+        # making partner wise payment
+        payment_ids = []
+        for partner in list(group_data):
+            # update active_ids with active invoice ids
+            if context.get("active_ids", False) and group_data[partner].get(
+                "inv_val", False
+            ):
+                context.update({"active_ids": list(group_data[partner]["inv_val"])})
+            payment = (
+                self.env["account.payment"]  # pylint: disable=context-overridden
+                .with_context(context)
+                .create(self.get_payment_values(group_data=group_data[partner]))
+            )
+            payment_ids.append(payment.id)
+            payment.action_post()
+            # Reconciliation
+            domain = [
+                ("account_type", "in", ("receivable", "payable")),
```

Comment:
Looks like `account.payment`'s `line_ids` in Odoo 16 refer to the `account.move.line`s of the inherited `account.move`. `account_type` on `account.move.line` is a related field on `account.account`'s `account_type`. These don't seem to be valid values for `account_type`. The values `receivable` and `payable` should be updated to reflect the valid values in this selection: https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_account.py#L44-L64

Reply:
https://github.com/OCA/account-payment/pull/690 does not have this issue BTW. Maybe go with that one?